src/lib/providers/TransferProvider.tsx

```diff
@@ -26,36 +35,52 @@ const TransferProvider: React.FC<TransferProviderType> = ({ children }) => {
 	const { addNotification } = useNotification();
 	const [transferring, setTransferring] = useState<TransferSubmitParams[]>([]);
 	const [transferred, setTransferred] = useState<TransferSubmitParams[]>([]);
-	const [
-		finishedTransferring,
-		setFinishedTransferring,
-	] = useState<TransferSubmitParams | null>(null);
 
-	const submit = async (params: TransferSubmitParams) => {
-		addNotification(`Started transfer`);
-		setTransferring([...transferring, params]);
+	const registryContract = useZnsContracts()!.registry;
+	//- Web3 Wallet Data
+	const walletContext = useWeb3React<Web3Provider>();
+	const { account } = walletContext;
+
+	const transferRequest = async (params: TransferSubmitParams) => {
+		if (!account) {
+			console.error('No wallet detected');
+			return;
+		}
+		if (account.toLowerCase() !== params.ownerId.toLowerCase()) {
+			console.error('You are not the owner');
+			return;
+		}
+
+		try {
+			const tx = await registryContract.transferFrom(
+				account,
+				params.walletAddress,
+				params.domainId,
+			);
 
-		// TODO: Back end transfer request.
-		const transfer = async () => {
-			setFinishedTransferring(params);
-		};
+			addNotification(`Started transfer`);
+			setTransferring([...transferring, params]);
 
-		await transfer();
-	};
+			tx.wait()
```

Comment:
Avoid using `.then` in this code base. Since we can use async/await functions it's best for us to only use async/await since mixing the two can easily cause bugs and confusion later on.

You can imitate this exact same logic by defining an async function here, then calling it, but not awaiting that call. (Take a look at src/providers/MintProvider.tsx and the `finishMinting` function that gets called if you'd like an example)